tests/test_routes.py

```diff
@@ -1,42 +1,87 @@
 """
-TestYourResourceModel API Service Test Suite
+Shopcarts API Service Test Suite
 
 Test cases can be run with the following:
-  nosetests -v --with-spec --spec-color
+  green
   coverage report -m
 """
-import os
 import logging
 from unittest import TestCase
 from unittest.mock import MagicMock, patch
+
 from service import app
-from service.models import db
 from service.common import status  # HTTP Status Codes
+from service.models import db, init_db, Shopcart
+from tests.factories import ShopcartFactory, ItemFactory
+from . import DATABASE_URI, BASE_URL
 
 
-######################################################################
-#  T E S T   C A S E S
-######################################################################
-class TestYourResourceServer(TestCase):
+class TestShopcartsService(TestCase):
     """ REST API Server Tests """
 
     @classmethod
     def setUpClass(cls):
         """ This runs once before the entire test suite """
+        app.config["TESTING"] = True
+        app.config["DEBUG"] = False
+        app.config["SQLALCHEMY_DATABASE_URI"] = DATABASE_URI
+        app.logger.setLevel(logging.CRITICAL)
+        init_db(app)
 
     @classmethod
     def tearDownClass(cls):
         """ This runs once after the entire test suite """
+        db.session.close()
 
     def setUp(self):
         """ This runs before each test """
         self.client = app.test_client()
+        db.session.query(Shopcart).delete()  # clean up the last tests
+        db.session.commit()
 
     def tearDown(self):
         """ This runs after each test """
+        db.session.remove()
+
+    ######################################################################
+    #  H E L P E R   M E T H O D S
+    ######################################################################
+
+    def _create_an_empty_shopcart(self):
```

Comment:
Do you think creating multiple shopcarts is better? Like professor's sample code: `def _create_accounts(self, count):
        """Factory method to create accounts in bulk"""
        accounts = []
        for _ in range(count):
            account = AccountFactory()
            resp = self.client.post(BASE_URL, json=account.serialize())
            self.assertEqual(
                resp.status_code,
                status.HTTP_201_CREATED,
                "Could not create test Account",
            )
            new_account = resp.get_json()
            account.id = new_account["id"]
            accounts.append(account)
        return accounts
`

Reply:
thank you for the suggestion, updated in the new commit 🙂